feat(RDS): add a data source to query tasks (#6846)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/rds_tasks.md

---
subcategory: "Relational Database Service (RDS)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_rds_tasks"
description: |-
  Use this data source to get the list of asynchronous tasks for a specified
  RDS instance within a specified time range.
---

# huaweicloud_rds_tasks

Use this data source to get the list of asynchronous tasks for a specified
RDS instance within a specified time range.

## Example Usage

```hcl
variable "instance_id" {}

data "huaweicloud_rds_tasks" "test" {
  instance_id = var.instance_id
  start_time  = "1747699200000"

> }

Reviewer: ```suggestion
variable "instance_id" {}
variable "start_time" {}

data "huaweicloud_rds_tasks" "test" {
  instance_id = var.instance_id
  start_time  = var.start_time
}
```

```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the resource.
  If omitted, the provider-level region will be used.

* `instance_id` - (Required, String) Specifies the ID of the RDS instance.

* `start_time` - (Required, String) Specified the start time in UTC timestamp format
  (miliseconds since epoch).

* `end_time` - (Optional, String) Specified the end time in UTC timestamp format

>   (miliseconds since epoch).

Reviewer: `miliseconds` is wrong spelling, should be `milliseconds`

  The [jobs](#jobs_struct) structure is documented below.

<a name="jobs_struct"></a>
The `jobs` block supports:

* `id` - Indicates the task ID.

* `name` - Indicates the task name.

* `status` - Indicates the task execution status.
  + **Running**: Indicates the task is being executed.
  + **Completed**: Indicates the task is successfully executed.
  + **Failed**: Indicates the task fails to be executed.

* `created` - Indicates the creation time. The value is in the **yyyy-mm-ddThh:mm:ssZ** format.
  T is the separator between the calendar and the hourly notation of time. Z indicates the time
  zone offset. For example, in the Beijing time zone, the time zone offset is shown as +0800.

* `ended` - Indicates the end time. The value is in the **yyyy-mm-ddThh:mm:ssZ** format. T is
  the separator between the calendar and the hourly notation of time. Z indicates the time
  zone offset. For example, in the Beijing time zone, the time zone offset is shown as +0800.

* `process` - Indicates the task execution progress. The execution progress (such as "60",
  indicating the task execution progress is 60%) is displayed only when the task is being
  executed. Otherwise, "" is returned.

> * `instance` - Indicates information of the DB instance on which the task is executed.

Reviewer: ```suggestion
* `instance` - Indicates information of the DB instance on which the task is executed.

  The [instance](#jobs_instance_struct) structure is documented below.
```

* `task_detail` - Indicates the displayed information varies depending on the tasks.

* `fail_reason` - Indicates the error information displayed when a task failed.

* `entities` - Indicates the displayed information varies depending on the tasks.

> The 'instance' block within 'jobs' supports:

Reviewer: ```suggestion
<a name="jobs_instance_struct"></a>
The `instance` block supports:
```

---
subcategory: "Relational Database Service (RDS)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_rds_tasks"
description: |-
  Use this data source to get the list of asynchronous tasks for a specified
  RDS instance within a specified time range.
---

# huaweicloud_rds_tasks

Use this data source to get the list of asynchronous tasks for a specified
RDS instance within a specified time range.

## Example Usage

```hcl
variable "instance_id" {}
variable "start_time" {}

data "huaweicloud_rds_tasks" "test" {
  instance_id = var.instance_id
  start_time  = var.start_time
}
```

## Argument Reference

The following arguments are supported:

> * `region` - (Optional, String) Specifies the region in which to query the resource.

Reviewer: ```suggestion
* `region` - (Optional, String) Specifies the region in which to query the data source.
```

The `jobs` block supports:

* `id` - Indicates the task ID.

* `name` - Indicates the task name.

* `status` - Indicates the task execution status.
  + **Running**: Indicates the task is being executed.
  + **Completed**: Indicates the task is successfully executed.
  + **Failed**: Indicates the task fails to be executed.

* `created` - Indicates the creation time. The value is in the **yyyy-mm-ddThh:mm:ssZ** format.
  T is the separator between the calendar and the hourly notation of time. Z indicates the time
  zone offset. For example, in the Beijing time zone, the time zone offset is shown as +0800.

* `ended` - Indicates the end time. The value is in the **yyyy-mm-ddThh:mm:ssZ** format. T is
  the separator between the calendar and the hourly notation of time. Z indicates the time
  zone offset. For example, in the Beijing time zone, the time zone offset is shown as +0800.

* `process` - Indicates the task execution progress. The execution progress (such as "60",
  indicating the task execution progress is 60%) is displayed only when the task is being
  executed. Otherwise, "" is returned.

* `instance` - Indicates information of the DB instance on which the task is executed.

> The [instance](#jobs_instance_struct) structure is documented below.

Reviewer: ```suggestion

  The [instance](#jobs_instance_struct) structure is documented below.
```

layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_rds_tasks"
description: |-
  Use this data source to get the list of asynchronous tasks for a specified
  RDS instance within a specified time range.
---

# huaweicloud_rds_tasks

Use this data source to get the list of asynchronous tasks for a specified
RDS instance within a specified time range.

## Example Usage

```hcl
variable "instance_id" {}
variable "start_time" {}

data "huaweicloud_rds_tasks" "test" {
  instance_id = var.instance_id
  start_time  = var.start_time
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the data source.
  If omitted, the provider-level region will be used.

* `instance_id` - (Required, String) Specifies the ID of the RDS instance.

* `start_time` - (Required, String) Specified the start time in UTC timestamp format
  (milliseconds since epoch).

* `end_time` - (Optional, String) Specified the end time in UTC timestamp format
  (milliseconds since epoch).

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> 

Reviewer 2: the id of the data source is needed
